### ISSUE TITLE
[Bug]: Standalone pod status becomes error when running `test_e2e.py` at the step `get_flush_state` without any chaos

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#21209]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20221213-18d546e6
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus==2.3.0.dev21
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2022-12-13T21:09:44.866Z] + kubectl get pods -o wide

[2022-12-13T21:09:44.866Z] + grep standalone-pod-failure-510

[2022-12-13T21:09:44.866Z] standalone-pod-failure-510-etcd-0                               1/1     Running           0              5m19s   10.102.9.72     devops-node13   <none>           <none>

[2022-12-13T21:09:44.866Z] standalone-pod-failure-510-milvus-standalone-85d8d8b8bb-nqt8d   0/1     Error             3 (67s ago)    5m19s   10.102.9.75     devops-node13   <none>           <none>

[2022-12-13T21:09:44.866Z] standalone-pod-failure-510-minio-79b4c7d77-bqhcp                1/1     Running           0              5m19s   10.102.9.71     devops-node13   <none>           <none>
```

```
[2022-12-13T21:06:36.254Z] [2022-12-13 21:06:35 - INFO - ci_test]: [test][2022-12-13T21:06:35Z] [0.00426441s] e2e__M7ebuYNB flush -> None (wrapper.py:30)

[2022-12-13T21:06:36.255Z] [2022-12-13 21:06:35 - INFO - ci_test]: assert create collection: 0.023760318756103516, init_entities: 0 (test_e2e.py:24)

[2022-12-13T21:06:36.255Z] [2022-12-13 21:06:36 - INFO - ci_test]: [test][2022-12-13T21:06:35Z] [0.30270448s] e2e__M7ebuYNB insert -> (insert count: 3000, delete count: 0, upsert count: 0, timestamp: 438033605250777089, success count: 3000, err count: 0) (wrapper.py:30)

[2022-12-13T21:06:36.255Z] [2022-12-13 21:06:36 - INFO - ci_test]: assert insert: 0.30307722091674805 (test_e2e.py:31)

[2022-12-13T21:06:37.183Z] [2022-12-13 21:06:36 - WARNING - pymilvus.decorators]: [get_flush_state] retry:4, cost: 0.27s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:06:37.183Z] [2022-12-13 21:06:37 - WARNING - pymilvus.decorators]: [get_flush_state] retry:5, cost: 0.81s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:06:38.114Z] [2022-12-13 21:06:37 - WARNING - pymilvus.decorators]: [get_flush_state] retry:6, cost: 2.43s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:06:40.631Z] [2022-12-13 21:06:40 - WARNING - pymilvus.decorators]: [get_flush_state] retry:7, cost: 7.29s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:06:48.719Z] [2022-12-13 21:06:47 - WARNING - pymilvus.decorators]: [get_flush_state] retry:8, cost: 21.87s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:07:10.602Z] [2022-12-13 21:07:09 - WARNING - pymilvus.decorators]: [get_flush_state] retry:9, cost: 60.00s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:08:18.224Z] [2022-12-13 21:08:09 - WARNING - pymilvus.decorators]: [get_flush_state] retry:10, cost: 60.00s, reason: <_MultiThreadedRendezvous: StatusCode.UNAVAILABLE, failed to connect to all addresses> (decorators.py:71)

[2022-12-13T21:09:14.393Z] [2022-12-13 21:09:09 - ERROR - pymilvus.decorators]: RPC error: [get_flush_state], <MilvusUnavailableException: (code=1, message=server Unavailable: Retry run out of 10 retry times)>, <Time:{'RPC start': '2022-12-13 21:06:36.740409', 'RPC error': '2022-12-13 21:09:09.823350'}> (decorators.py:108)

[2022-12-13T21:09:14.393Z] [2022-12-13 21:09:09 - ERROR - pymilvus.decorators]: RPC error: [flush], <MilvusUnavailableException: (code=1, message=server Unavailable: Retry run out of 10 retry times)>, <Time:{'RPC start': '2022-12-13 21:06:36.236459', 'RPC error': '2022-12-13 21:09:09.823853'}> (decorators.py:108)

[2022-12-13T21:09:14.393Z] [2022-12-13 21:09:09 - ERROR - ci_test]: Traceback (most recent call last):

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler

[2022-12-13T21:09:14.393Z]     return func(self, *args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 775, in get_flush_state

[2022-12-13T21:09:14.393Z]     response = future.result()

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 744, in result

[2022-12-13T21:09:14.393Z]     raise self

[2022-12-13T21:09:14.393Z] grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:

[2022-12-13T21:09:14.393Z] 	status = StatusCode.UNAVAILABLE

[2022-12-13T21:09:14.393Z] 	details = "failed to connect to all addresses"

[2022-12-13T21:09:14.393Z] 	debug_error_string = "{"created":"@1670965749.822761222","description":"Failed to pick subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":3260,"referenced_errors":[{"created":"@1670965749.822759933","description":"failed to connect to all addresses","file":"src/core/lib/transport/error_utils.cc","file_line":167,"grpc_status":14}]}"

[2022-12-13T21:09:14.393Z] >

[2022-12-13T21:09:14.393Z] 

[2022-12-13T21:09:14.393Z] The above exception was the direct cause of the following exception:

[2022-12-13T21:09:14.393Z] 

[2022-12-13T21:09:14.393Z] Traceback (most recent call last):

[2022-12-13T21:09:14.393Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper

[2022-12-13T21:09:14.393Z]     res = func(*args, **_kwargs)

[2022-12-13T21:09:14.393Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request

[2022-12-13T21:09:14.393Z]     return func(*arg, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 280, in flush

[2022-12-13T21:09:14.393Z]     conn.flush([self.name], timeout=timeout, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler

[2022-12-13T21:09:14.393Z]     raise e

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler

[2022-12-13T21:09:14.393Z]     return func(*args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2022-12-13T21:09:14.393Z]     ret = func(self, *args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler

[2022-12-13T21:09:14.393Z]     raise e

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler

[2022-12-13T21:09:14.393Z]     return func(self, *args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 834, in flush

[2022-12-13T21:09:14.393Z]     _check()

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 822, in _check

[2022-12-13T21:09:14.393Z]     self._wait_for_flushed(segment_ids)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 796, in _wait_for_flushed

[2022-12-13T21:09:14.393Z]     flush_ret = self.get_flush_state(segment_ids, timeout, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler

[2022-12-13T21:09:14.393Z]     raise e

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler

[2022-12-13T21:09:14.393Z]     return func(*args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2022-12-13T21:09:14.393Z]     ret = func(self, *args, **kwargs)

[2022-12-13T21:09:14.393Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 66, in handler

[2022-12-13T21:09:14.393Z]     raise MilvusUnavailableException(message=f"server Unavailable: {timeout_msg}") from e

[2022-12-13T21:09:14.393Z] pymilvus.exceptions.MilvusUnavailableException: <MilvusUnavailableException: (code=1, message=server Unavailable: Retry run out of 10 retry times)>

[2022-12-13T21:09:14.393Z]  (api_request.py:39)

[2022-12-13T21:09:14.393Z] [2022-12-13 21:09:09 - ERROR - ci_test]: (api_response) : <MilvusUnavailableException: (code=1, message=server Unavailable: Retry run out of 10 retry times)> (api_request.py:40)
```

```
[2022/12/13 21:09:38.725 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:411] ["insertBufferNode syncing BufferData"] [segmentID=438033598259596620] [flushed=false] [dropped=false] [auto=false] [position="channel_name:\"by-dev-rootcoord-dml_0_438033598259396611v0\" msgID:\"\\r\\000,\\235<5\\024\\006\" msgGroup:\"by-dev-dataNode-4-by-dev-rootcoord-dml_0_438033598259396611v0\" timestamp:438033605250777094 "] [channel=by-dev-rootcoord-dml_0_438033598259396611v0]
[2022/12/13 21:09:38.726 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:273] ["segment buffer status"] [no.=0] [segmentID=438033598259596621] [channel=by-dev-rootcoord-dml_1_438033598259396611v1] [size=1497] [limit=21077]
[2022/12/13 21:09:38.726 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:365] ["merging sync task, updating flushed flag"] [segmentID=438033598259596621] [flushed=false] [dropped=false]
[2022/12/13 21:09:38.726 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:411] ["insertBufferNode syncing BufferData"] [segmentID=438033598259596621] [flushed=false] [dropped=false] [auto=false] [position="channel_name:\"by-dev-rootcoord-dml_1_438033598259396611v1\" msgID:\"\\016\\000,\\235<5\\024\\006\" msgGroup:\"by-dev-dataNode-4-by-dev-rootcoord-dml_1_438033598259396611v1\" timestamp:438033605250777094 "] [channel=by-dev-rootcoord-dml_1_438033598259396611v1]
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7fea8d92df89]

runtime stack:
runtime.throw({0x3644dfa?, 0x7fea8da7ebe0?})
	/usr/local/go/src/runtime/panic.go:992 +0x71
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:802 +0x3a9

goroutine 1110 [syscall]:
runtime.cgocall(0x2ca8ac0, 0xc001c6af68)
	/usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc001c6af40 sp=0xc001c6af08 pc=0x119eb7c
github.com/milvus-io/milvus/internal/storage._Cfunc_ReleasePayloadWriter(0x7fe84731c000)
	_cgo_gotypes.go:542 +0x45 fp=0xc001c6af68 sp=0xc001c6af40 pc=0x242dc45
github.com/milvus-io/milvus/internal/storage.(*PayloadWriter).ReleasePayloadWriter.func1(0x4ea93c8?)
	/go/src/github.com/milvus-io/milvus/internal/storage/payload.go:356 +0x3a fp=0xc001c6afa0 sp=0xc001c6af68 pc=0x242f91a
github.com/milvus-io/milvus/internal/storage.(*PayloadWriter).ReleasePayloadWriter(0x32a2720?)
	/go/src/github.com/milvus-io/milvus/internal/storage/payload.go:356 +0x19 fp=0xc001c6afb8 sp=0xc001c6afa0 pc=0x242f8b9
github.com/milvus-io/milvus/internal/storage.(*baseEventWriter).Close(...)
	/go/src/github.com/milvus-io/milvus/internal/storage/event_writer.go:162
github.com/milvus-io/milvus/internal/storage.(*InsertCodec).Serialize(0xc000a1f528, 0xc001c6b448?, 0x67?, 0xc004d43a60)
	/go/src/github.com/milvus-io/milvus/internal/storage/data_codec.go:429 +0x1271 fp=0xc001c6b1e0 sp=0xc001c6afb8 pc=0x240ab91
github.com/milvus-io/milvus/internal/datanode.(*rendezvousFlushManager).flushBufferData(0xc004801080, 0xc004d4ecc0, 0x614353c9ff30d4c, 0x20?, 0xa4?, 0xc001c6b868?)
	/go/src/github.com/milvus-io/milvus/internal/datanode/flush_manager.go:367 +0x2dc fp=0xc001c6b688 sp=0xc001c6b1e0 pc=0x26b161c
github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Sync.func1()
	/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:422 +0x63 fp=0xc001c6b760 sp=0xc001c6b688 pc=0x26aa143
github.com/milvus-io/milvus/internal/util/retry.Do({0x39f9290, 0xc0054a47c0}, 0xc001c6b9d8, {0xc001c6b980, 0x1, 0x2000?})
	/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:37 +0x103 fp=0xc001c6b8f8 sp=0xc001c6b760 pc=0x1da8f23
github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Sync(0xc0015ce700, 0xc004d58750?, {0xc004d58750?, 0xc0043e7b00?, 0xc0043e7aa0?}, 0xc0043e7b00)
	/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:421 +0x729 fp=0xc001c6bc30 sp=0xc001c6b8f8 pc=0x26a98a9
github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Operate(0xc0015ce700, {0xc00527d180?, 0x43?, 0x36bbaa8?})
	/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:171 +0x6b7 fp=0xc001c6be90 sp=0xc001c6bc30 pc=0x26a6a37
github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).work(0xc0053ff2c0)
	/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:127 +0x2b8 fp=0xc001c6bfc8 sp=0xc001c6be90 pc=0x24eb818
github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).Start.func1()
	/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:70 +0x26 fp=0xc001c6bfe0 sp=0xc001c6bfc8 pc=0x24eb246
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc001c6bfe8 sp=0xc001c6bfe0 pc=0x1207e61
created by github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).Start
	/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:70 +0x79

goroutine 1 [chan receive]:
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc00150fe58, 0x1, {0xc000466090?, 0xe?})
	/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:304 +0xc31
github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc00049c060, {0xc0001bc060?, 0x3, 0x3}, 0xc0004a8000)
	/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:117 +0x6e5
github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc0001bc060?, 0x3, 0x3})
	/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e
main.main()
	/go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e

goroutine 337 [select]:
github.com/milvus-io/milvus/internal/config.refresher.refreshPeriodically({0x2540be400, 0xc000748000, {0x0, {0x1, 0x0}}, {0x0, 0x0}, 0xc00129a0a0}, {0x35dec9b, 0xa})
	/go/src/github.com/milvus-io/milvus/internal/config/refresher.go:59 +0x1aa
created by github.com/milvus-io/milvus/internal/config.refresher.start.func1
	/go/src/github.com/milvus-io/milvus/internal/config/refresher.go:46 +0xea

goroutine 147 [chan receive]:
github.com/panjf2000/ants/v2.(*Pool).purgePeriodically(0xc0008fdf10)
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:69 +0x8b
created by github.com/panjf2000/ants/v2.NewPool
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:137 +0x34a

goroutine 274 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc00127f3b0, 0x1)
	/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/controlbuf.go:407 +0x115
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc0004a9d40)
	/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/controlbuf.go:534 +0x85
google.golang.org/grpc/internal/transport.newHTTP2Client.func3()
	/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/http2_client.go:415 +0x65
created by google.golang.org/grpc/internal/transport.newHTTP2Client
	/go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/http2_client.go:413 +0x1f91
....
```

### Expected Behavior

all test cases passed

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-cron/detail/chaos-test-cron/510/pipeline
log:
[artifacts-standalone-pod-failure-510-server-logs (1).tar.gz](https://github.com/milvus-io/milvus/files/10223806/artifacts-standalone-pod-failure-510-server-logs.1.tar.gz)

[artifacts-standalone-pod-failure-510-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/10223807/artifacts-standalone-pod-failure-510-pytest-logs.tar.gz)

**Comment by smellthemoon** [comment]
fixed with #21218

**Comment by zhuwenxing** [comment]
Not reproduced in `master-20221216-e05ac562`
